src/test/scala/nl/knaw/dans/lib/encode/PathEncodingSpec.scala

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib.encode
+
+import java.nio.file.Paths
+
+import org.scalatest.{ FlatSpec, Matchers, OptionValues }
+
+class PathEncodingSpec extends FlatSpec with Matchers with OptionValues {
+
+  "escapePath" should "return a string without any escaped characters when the path segments contain only alphanumeric characters or a '_' character" in {
+    val path = Paths.get("abcdefghijklmnopqrstuvwxyz/ABCDEFGHIJKLMNOPQRSTUVWXYZ/1234567890/_")
+    path.escapePath shouldBe "abcdefghijklmnopqrstuvwxyz/ABCDEFGHIJKLMNOPQRSTUVWXYZ/1234567890/_"
+  }
+
+  it should "return a string where all the characters, except alphanumeric characters and characters '_' and '/', are escaped" in {
+    val path = Paths.get("abc/def_12345/! @#$%^&*()-/[]{}'|`~,.<>?/好棒啊，汉语也可以")
```

Comment:
for those reviewers who are interested...

https://translate.google.com/?source=osdd#view=home&op=translate&sl=zh-CN&tl=nl&text=%E5%A5%BD%E6%A3%92%E5%95%8A%EF%BC%8C%E6%B1%89%E8%AF%AD%E4%B9%9F%E5%8F%AF%E4%BB%A5

Reply:
I vaguely remember warnings about code with non-ASCII characters, but don't remember the context. It might be a problem on systems that somehow don't read these files with UTF-8 encoding.

Reply:
@jo-pol: maybe something with Greek of Hebrew file names :0) ? The risk of putting Chinese chars in your source code is that someone else might have a compiler that expects a different source file encoding than the one you have saved it in. To avoid that you should use spell out the unicode code points with `\u`-sequences. 

See: https://stackoverflow.com/questions/4448180/why-does-java-permit-escaped-unicode-characters-in-the-source-code

There seems to be a tool called `native2ascii` that will do the conversion for you.

Reply:
The point of having those Chinese characters here is to show that they are escaped, just as all the other non-numeric characters in the string.  It kind of defeats the whole idea if we have unicode points instead of 'visible' characters.